### DISCUSSION TITLE
How to create Synchro Control projects using create iTwin API?

### DISCUSSION THREAD

**Discussion by habdirad** [iTwin/itwinjs-core#6178] [Q&A] [answered]
Hello, 
My list of Synchro projects shows in the list of iTwins that I extract with iTwins GET https://developer.bentley.com/apis/itwins/operations/get-my-itwins/, and those projects have the following properties class":"Endeavor","subClass":"Project","type":null. 

However, when I create new iTwins with the same properties using Create iTwin APIs, it is creating iTwins that do not show in Synchro Control. https://developer.bentley.com/apis/itwins/operations/create-itwin/

Question: Is there a way to create Synchro control projects with iTwin APIs?
@ben-polinsky 

Thank you

**Comment by pratikshan85** [comment]
This is the expected behavior. Currently, Bentley doesn't support BIC((Bentley infrastructure cloud ) project to be created using iTwinplantform (iTwin API).

**Reply by cmonk28** [reply]
So in the example of https://developer.bentley.com/apis/itwins/operations/create-itwin/,  the example creates a project with the following properties class":"Endeavor","subClass":"Project".

But in both the get-my-itwins and get-itwins api calls... the "Endeavor" class isn't supported.  So you can create an Endeavor project, but you can't get it???   Is there any further support or effort on increasing API calls and its support for Endeavor class projects?

**Reply by T1St1** [reply]
Right now BIC only supports projects created through the portal. New ITwin APIs are not supported yet and will not be for some time.

**Reply by davidhjones** [reply, marked as answer]
Fundamentally, there are no "BIC projects" and "iTwin Platform projects" -- there are no differences between the two iTwins. 

Synchro (and BIC) limits the UX in regards to the iTwins (aka projects) that are available in their interface. This application does not allow for you to access the iTwins that have been created using the iTwin APIs. This needs to be fixed in BIC/synchro, to be more inclusive, allowing for iTwins created outside their application to be used within it.

**Reply by davidhjones** [reply]
> So in the example of https://developer.bentley.com/apis/itwins/operations/create-itwin/, the example creates a project with the following properties class":"Endeavor","subClass":"Project".
> 
> But in both the get-my-itwins and get-itwins api calls... the "Endeavor" class isn't supported. So you can create an Endeavor project, but you can't get it??? Is there any further support or effort on increasing API calls and its support for Endeavor class projects?

What do you mean you can't get it? Can you provide more details on what you cannot get? Exactly what API call are you making?

**Reply by cmonk28** [reply]
Is there a way to get the repository or listing of documents within an Endeavor class project?  Is there a way to get it for All Projects (including Endeavor).
Thanks,

**Reply by davidhjones** [reply]
> Is there a way to get the repository or listing of documents within an Endeavor class project? Is there a way to get it for All Projects (including Endeavor). Thanks,

I'm not sure I understand your question. There is a Storage API, passing in an iTwin Id, you can fetch all of the files for an iTwin: https://developer.bentley.com/apis/storage/overview/

Your previous question was in regards to the results of the get-my-itwins and get-itwins api calls. Did you resolve those issues?

**Reply by pratikshan85** [reply]
To get the all Projects (including Endeavor) you can call the API :
[Get my iTwins](https://developer.bentley.com/apis/itwins/operations/get-my-itwins/#get-my-itwins)
GET https://api.bentley.com/itwins/?subClass[&status][&type][&number][&displayName][&parentId][&iTwinAccountId][&$search][&$top][&$skip][&includeInactive]

**Reply by cmonk28** [reply]
Thanks Pratikshan85, this will list out the list of projects, this is fine.

Specifically davidhjones, trying the storage API I believe is what i'm looking for. Entering a projectid (that i san Endeavor class) under the api call _GET https://api.bentley.com/storage/?projectId[&$top][&$skip]_  are their specific roles/permissions required, as I currently get a 403 Forbidden error returning.

Thanks

**Reply by davidhjones** [reply]
> Thanks Pratikshan85, this will list out the list of projects, this is fine.
> 
> Specifically davidhjones, trying the storage API I believe is what i'm looking for. Entering a projectid (that i san Endeavor class) under the api call _GET [https://api.bentley.com/storage/?projectId[&$top][&$skip]](https://api.bentley.com/storage/?projectId%5B&$top%5D%5B&$skip%5D)_ are their specific roles/permissions required, as I currently get a 403 Forbidden error returning.
> 
> Thanks

I see @cmonk28 -- I would suggest you create a new discussion on this topic, and we can try to help you with your issue.
